Highlight quote after scrolling to it

PVince81 opened the pull request:
Fixes https://github.com/nextcloud/spreed/issues/4516

When clicking on a quote, we scroll to it and now there's a new
highlight animation that fades out

The animation and scrolling logic was moved from the Quote vue to the
Message vue. The Quote vue will now only send an event to tell that we
want to focus a specific message.

A DOM-level method was used for triggering the highlight to be able to
access it with document.getElementById() as it's cheaper than trying to
look up the message by id within the collection of MessagesList.

If the latter is not wanted we should then rework the messages structure inside of MessagesList to have the messages in an associative array with message id as key. Or build yet another index-like structure to be able to find the messages. So far for this use case it feels a bit like overkill.

nickvergessen commented on src/components/MessagesList/MessagesGroup/Message/Message.vue:
```diff
@@ -439,4 +461,14 @@ export default {
+
+// TODO: define these colors globally (these are borrowed from the files app upload.css)
+@keyframes highlight-animation {
+	0% { background-color: rgba(255, 255, 140, 1); }
```
Need to introduce scss vars in server so it's themed somehow and dark aware

PVince81 replied:
I now simply switched to the existing hover variable

PVince81 commented:
More improvements:
- switched to existing hover variable
- increased animation time
- fixed "click on quote in new message form"
- enhancement: when coming from search, if the URL anchor contains a message id, it is now also highlighted!

nickvergessen commented on src/components/MessagesList/MessagesGroup/Message/Message.vue:
```diff
@@ -439,4 +461,14 @@ export default {
 	border-radius: 8px;
 }
+
+.highlight-animation {
```
missing the border-radius compared to the hover?

PVince81 replied:
I didn't intend to add one, but you're right that it might look better.
I'll add it

PVince81 replied:
Added

nickvergessen reviewed (changes requested):
Looks good apart from the missing border radius

PVince81 commented:
Now that https://github.com/nextcloud/spreed/issues/4592 is merged the behavior might change slightly.
It is possible that I'll need to add extra handling since now we have a scenario where the page doesn't reload.

- [ ] recheck "click search result for the current conversation" behavior

PVince81 commented:
I'm not done yet... Something is fishy: it should be possible to smooth scroll despite the "browser's anchor logic teleports to location before we get a chance to smooth scroll"

PVince81 commented:
I managed to improve it a bit with a tiny delay hack.

At least now when clicking the search result the entry will end up in the middle of the page, even though the scroll direction might be wrong. The best fix would be to add a "preventDefault" somehow on the search result link... Something for another time once I get to study a bit how the latter works.
